exporter/metrics_test.go

```diff
@@ -2,14 +2,98 @@ package exporter
 
 import (
 	"testing"
+	"time"
 
-	"gotest.tools/assert"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson/primitive"
 )
 
-func TestMakeRawMetricName(t *testing.T) {
-	for name, fqName := range map[string]string{
-		"dd.serverStatus.opcountersRepl.delete": "dd_server_status_opcounters_repl_delete",
-	} {
-		assert.Equal(t, fqName, makeRawMetricName(name))
+// Test metric renaming and labeling.
+func TestMetricRename(t *testing.T) {
+	tcs := []struct {
+		prefix     string
+		name       string
+		wantMetric string
+		wantLabel  string
+	}{
+		{
+			prefix:     "serverStatus.metrics.commands.saslStart.",
+			name:       "total",
+			wantMetric: "mongodb_ss_metrics_commands_saslStart_total",
+		},
+		{
+			prefix:     "serverStatus.metrics.commands._configsvrShardCollection.",
+			name:       "failed",
+			wantMetric: "mongodb_ss_metrics_commands_configsvrShardCollection_failed",
+		},
+		{
+			prefix:     "serverStatus.wiredTiger.lock.",
+			name:       "metadata lock acquisitions",
+			wantMetric: "mongodb_ss_wt_lock_metadata_lock_acquisitions",
+		},
+		{
+			prefix:     "serverStatus.wiredTiger.perf.",
+			name:       "file system write latency histogram (bucket 5) - 500-999ms",
+			wantMetric: "mongodb_ss_wt_perf",
+			wantLabel:  "perf_bucket",
+		},
+		{
+			prefix:     "serverStatus.wiredTiger.transaction.",
+			name:       "rollback to stable updates removed from lookaside",
+			wantMetric: "mongodb_ss_wt_txn_rollback_to_stable_updates_removed_from_lookaside",
+		},
 	}
+
+	for _, tc := range tcs {
+		metric, label := rename(tc.prefix, tc.name)
+		assert.Equal(t, tc.wantMetric, metric, tc.prefix+tc.name)
+		assert.Equal(t, tc.wantLabel, label, tc.prefix+tc.name)
+	}
+}
+
+// Test supported value types conversion.
+func TestMakeRawMetric(t *testing.T) {
+	prefix := "serverStatus.transactions."
+	name := "retriedCommandsCount"
+	testCases := []struct {
+		value     interface{}
+		wantVal   *float64
+		wantError bool
+	}{
+		{value: true, wantVal: pointerTo(1), wantError: false},
+		{value: false, wantVal: pointerTo(0), wantError: false},
+		{value: int32(1), wantVal: pointerTo(1), wantError: false},
+		{value: int64(2), wantVal: pointerTo(2), wantError: false},
+		{value: float32(1.23), wantVal: pointerTo(float64(float32(1.23))), wantError: false},
```

Comment:
it looks really weird.

Reply:
Why?

Reply:
casting 1.23 to float32 then to float64 and then to pointer

Reply:
We use https://github.com/AlekSi/pointer everywhere, so why not use it there?

Reply:
I haven't imported it because I only needed  a pointer to float 64 and only for testing so I didn't want to add a dependency for a 3 lines function.

Reply:
> I only needed a pointer to float 64

That's a story of this package in a nutshell :) I wrote similar functions in like 10 projects before extracting them

Reply:
@BupycHuk The conversion of interface{} that receives a float32 and float 64 is not the same.
If I don't force the conversion the test fails with 
```
expected: &prometheus.constMetric{desc:(*prometheus.Desc)(0xc000238380), valType:3, val:1.23, labelPairs:[]*io_prometheus_client.LabelPair(nil)}
                                actual  : &prometheus.constMetric{desc:(*prometheus.Desc)(0xc0002385b0), valType:3, val:1.2300000190734863, labelPairs:[]*io_prometheus_client.LabelPair(nil)}
```